helixbass/tsc-rust
# Type-check generic interface variable declaration

In this PR:
- type-check a variable declaration for a generic interface + object literal initializer

To test:
If you run `cargo run tmp.tsx` against a source file containing eg:
```
interface Foo<T> {
  bar: T
}

const a: Foo<number> = {
  bar: true,
}
```
then you should see a diagnostic debug-logged with `message_text: "Type 'boolean' is not assignable to type 'number'."`

If you run `cargo run tmp.tsx` against a source file containing eg:
```
interface Foo<T> {
  bar: T
}

const a: Foo<number> = {
  bar: 1,
}
```
then you should see no diagnostics debug-logged
